### PR TITLE
test: adding tests for @sec-tester/runner

### PR DESCRIPTION
**Pull request by Kif-Fando**

### CODE REVIEW

File: src/test/common_files.spec.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+
+import { SecRunner, SecScan } from '@sec-tester/runner'
+import { TestType, Severity } from '@sec-tester/scan'
+import { Configuration } from "@sec-tester/core";
+
+describe('COMMON_FILES', () => {
```
Comment (derevnjuk) [inline src/test/common_files.spec.ts:6]: Follow naming convention:

```ts
describe('SUT, e.g. /api', () => {
  describe('METHOD /path, .e.g POST /users', () => {
    it.todo('should do something, e.g. "should create a user" or "should not expose common files"');
  })
})
```

File: src/test/common_files.spec.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  beforeEach(async () => {
+    runner = new SecRunner(configuration);
+    await runner.init();
+
+  });
+
+  afterEach(async () => {
+    await runner.clear();
+  });
```
Comment (derevnjuk) [inline src/test/common_files.spec.ts:22]: Get rid of redundant async/await statements and brackets:
```suggestion
  afterEach(() => runner.clear());
```

File: src/test/common_files.spec.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  it('COMMON_FILES', async () => {
+    scan = runner.createScan({ tests: [TestType.COMMON_FILES, TestType.COOKIE_SECURITY, TestType.CSRF, ], name: 'COMMON_FILES' })
```
Comment (derevnjuk) [inline src/test/common_files.spec.ts:25]: I don't see any reason to have this local variable

File: src/test/common_files.spec.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    .timeout(3000000);
+  await scan.run({
+    method: 'GET',
+    url: `${process.env.URL}`
+  })
```
Comment (derevnjuk) [inline src/test/common_files.spec.ts:30]:
```suggestion
  return runner.createScan({ tests: [TestType.COMMON_FILES, TestType.COOKIE_SECURITY, TestType.CSRF, ], name: 'COMMON_FILES' })
    .timeout(3000000)
    .run({
      method: 'GET',
      url: `${process.env.URL}`
    })
```

File: src/test/common_files.spec.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  afterEach(async () => {
+    await runner.clear();
+  });
+
+  it('COMMON_FILES', async () => {
+    scan = runner.createScan({ tests: [TestType.COMMON_FILES, TestType.COOKIE_SECURITY, TestType.CSRF, ], name: 'COMMON_FILES' })
+    .timeout(3000000);
+  await scan.run({
+    method: 'GET',
+    url: `${process.env.URL}`
```
Comment (derevnjuk) [inline src/test/common_files.spec.ts:29]: You have to run tests against the local target

File: src/test/common_files.spec.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+
+import { SecRunner, SecScan } from '@sec-tester/runner'
+import { TestType, Severity } from '@sec-tester/scan'
+import { Configuration } from "@sec-tester/core";
+
+describe('COMMON_FILES', () => {
+  let runner: SecRunner;
+  let scan: SecScan;
+   let configuration: Configuration = new Configuration({
+    hostname: process.env.BRIGHT_CLUSTER
```
Comment (derevnjuk) [inline src/test/common_files.spec.ts:10]: To use env variables, you have to adjust jest config by setting `setupFile`

**Review by derevnjuk — CHANGES_REQUESTED** [review]
There are a certain number of formatting mistakes and oddities. Please run formatting and lining before pushing into the remote branch

File: src/test/common_files.spec.ts (hunk 2 above)
Comment (Kif-Fando) [reply src/test/common_files.spec.ts:22]: Wrote according to the documentation 😜:
https://github.com/NeuraLegion/sec-tester-js#:~:text=afterEach(async%20()%20%3D%3E%20%7B%0A%20%20%20%20await%20runner.clear()%3B%0A%20%20%7D)%3B

File: src/test/osi.spec.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (derevnjuk) [inline src/test/osi.spec.ts:1]: Redundant blank line

File: src/test/osi.spec.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import { TestType } from '@sec-tester/scan';
+
+describe('/api', () => {
+  let runner: SecRunner;
+  let scan: SecScan;
+
+  beforeEach(async () => {
+    runner = new SecRunner({ hostname: process.env.BRIGHT_CLUSTER });
+    await runner.init();
+
```
Comment (derevnjuk) [inline src/test/osi.spec.ts:12]: ditto

**Review by derevnjuk — COMMENTED** [review]
Please run `prettier` against tests

File: src/test/full_path_disclosure.spec.ts [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    it('should not contain errors that include full webroot path', () => {
+      return runner
+        .createScan({
+          tests: [TestType.FULL_PATH_DISCLOSURE],
+          name: 'FULL_PATH_DISCLOSURE',
+        })
+        .timeout(3000000)
+        .run({
+          method: 'GET',
+          url: `${process.env.SEC_TESTER_TARGET}/api/auth/oidc-client?`,
```
Comment (derevnjuk) [inline src/test/full_path_disclosure.spec.ts:39]:
```suggestion
          url: `${process.env.SEC_TESTER_TARGET}/api/auth/oidc-client`,
```

File: src/test/full_path_disclosure.spec.ts [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+    it('should not contain errors that include full webroot path', () => {
+      return runner
+        .createScan({
+          tests: [TestType.FULL_PATH_DISCLOSURE],
+          name: 'FULL_PATH_DISCLOSURE',
+        })
+        .timeout(3000000)
+        .run({
+          method: 'GET',
+          url: `${process.env.SEC_TESTER_TARGET}/api/users/one/undefined/adminpermission?`,
```
Comment (derevnjuk) [inline src/test/full_path_disclosure.spec.ts:53]:
```suggestion
          url: `${process.env.SEC_TESTER_TARGET}/api/users/one/undefined/adminpermission`,
```

File: src/test/ssti.spec.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+        .createScan({ tests: [TestType.SSTI], name: 'SSTI' })
+        .timeout(3000000)
+        .run({
+          method: 'POST',
+          headers: {
+            Accept: 'application/json, text/plain, */*',
+            'Content-Type': 'text/plain',
+            Origin: process.env.SEC_TESTER_TARGET,
+          },
+          body: `{{=""+1""}} {{=5589}} {{=55488}} {{=55}}{{=840931+350734}}`,
```
Comment (derevnjuk) [inline src/test/ssti.spec.ts:27]: It is better to use real payloads instead of vulnerable one

**Review by derevnjuk — CHANGES_REQUESTED** [review]

File: package.json [hunk 12]
```diff
@@ -30,6 +30,12 @@
     "@nestjs/core": "^7.6.7",
     "@nestjs/platform-fastify": "^7.6.7",
     "@nestjs/swagger": "^4.7.12",
+    "@sec-tester/bus": "^0.15.0",
+    "@sec-tester/core": "^0.15.0",
+    "@sec-tester/repeater": "^0.15.0",
+    "@sec-tester/reporter": "^0.15.0",
+    "@sec-tester/runner": "^0.15.0",
+    "@sec-tester/scan": "^0.15.0",
```
Comment (derevnjuk) [inline package.json:38]: Bump SecTester to the latest version

**Review by derevnjuk — APPROVED** [review]

File: src/test/auth.spec.ts [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+
+const generateToken = async (jwtType) => {
+  return await axios
+    .post(`${process.env.SEC_TESTER_TARGET}/api/auth/jwt/${jwtType}/login`, {
+      user: 'admin',
+      password: 'admin',
+      op: 'basic',
+    })
+    .then(({ headers }) => headers)
+    .then(({ authorization }) => authorization);
```
Comment (derevnjuk) [inline src/test/auth.spec.ts:13]: https://github.com/NeuraLegion/brokencrystals/pull/184#discussion_r885420553

**Review by derevnjuk — CHANGES_REQUESTED** [review]

**Review by derevnjuk — APPROVED** [review]

**Pull request merged**